go-controller/pkg/ovn/address_set.go

```diff
@@ -418,3 +420,48 @@ func getIPv4ASHashedName(name string) string {
 func getIPv6ASHashedName(name string) string {
 	return hashedAddressSet(name + ipv6AddressSetSuffix)
 }
+
+// This method checks if an address set exists in OVN NB or not. In some cases when a namespace
+// is created, there will be multiple callbacks to create addtess set and create/updates ACLs for
+// network polocies. To ensure address set is created under these circumstances, this method
+// will retry for 1 second to check if the address set exists.
+func addressSetExists(hashName string) (bool, error) {
+	err := utilwait.PollImmediate(100*time.Millisecond, 1*time.Second, func() (bool, error) {
+		uuid, stderr, err := util.RunOVNNbctl("--data=bare",
```

Comment:
Can you please use go binding instead? https://github.com/eBay/go-ovn/blob/master/client.go#L76

Reply:
I had thought about it but entire address set and policies are using the `RunOVNNbctl`commands. All the policy tests are also set up this way using fake-execs  which are  complex by the way. Hence I did it this way . Can we  please take up the task of converting this code and tests to use OVN go bindings as a separate task?

Reply:
Since we are polling here, it's going to be much better to leverage the go bindings. This is because the IDL keeps a cache of all of the objects in OVN DB, so the polling here will be much less expensive than issuing many ovn-nbctl cmds. You should be able to just mock out the client like we did for the logical port stuff. But before going down this path I would try to dig deeper into the race and see if it ends up that we are missing flows when a real address set and and policy ACL are configured out of order.